### PR TITLE
modify movie detail, add youtube player

### PR DESCRIPTION
**Pull request by tungnt0315**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -54,4 +54,8 @@ dependencies {
 
   compile 'com.android.support:recyclerview-v7:25.3.1'
   compile 'com.android.support:cardview-v7:25.3.1'
+
+  compile files('libs/YouTubeAndroidPlayerApi.jar')
+
```
Comment (daolq3012) [inline app/build.gradle]: pls remove blank line

File: app/build.gradle [hunk 2]
```diff
@@ -54,4 +54,8 @@ dependencies {
 
   compile 'com.android.support:recyclerview-v7:25.3.1'
   compile 'com.android.support:cardview-v7:25.3.1'
+
```
Comment (daolq3012) [inline app/build.gradle]: pls remove blank line

File: app/src/main/java/com/example/tung/moviedb_07/screen/movie_detail/MovieDetailActivity.java [hunk 3]
```diff
@@ -28,7 +27,9 @@ protected void onCreate(Bundle savedInstanceState) {
         Bundle bundle = getIntent().getExtras();
         Movie movie = bundle.getParcelable(BUNDLE_MOVIE);
         boolean isFavorite = bundle.getBoolean(BUNDLE_FAVORITE, false);
-        mViewModel = new MovieDetailViewModel(getApplicationContext(), movie, isFavorite);
+        Navigator navigator = new Navigator(this);
+        mViewModel =
+                new MovieDetailViewModel(movie, isFavorite, getSupportFragmentManager(), navigator);
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/screen/movie_detail/MovieDetailActivity.java]: pls remove param ``getSupportFragmentManager()`` from this method

File: app/src/main/java/com/example/tung/moviedb_07/screen/movie_detail/MovieDetailViewModel.java [hunk 4]
```diff
@@ -63,28 +66,90 @@ public String getProduction() {
...
+            String directorJob = mNavigator.getActivity().getString(R.string.label_director);
+            if (crew.getJob().equals(directorJob)) {
+                director = crew;
+                break;
+            }
+        } return director;
     }
 
+    public List<Cast> getCasts() {
+        return mMovie.getCredits().getCasts().subList(0, 10);
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/screen/movie_detail/MovieDetailViewModel.java]: pls check ``!=null``
like bellow
```java
if(a!=null && a.getb()!=null){
return a.getb().getc();
}
```

File: app/src/main/java/com/example/tung/moviedb_07/screen/movie_detail/MovieDetailViewModel.java [hunk 5]
```diff
@@ -63,28 +66,90 @@ public String getProduction() {
...
+
+    public HashtagView.TagsClickListener getGenreClickListener() {
+        return new HashtagView.TagsClickListener() {
+            @Override
+            public void onItemClicked(Object item) {
+                Bundle bundle = new Bundle();
+                bundle.putInt(Constant.BUNDLE_TAB, Constant.TAB_SEARCH_BY_GENRE);
+                bundle.putInt(Constant.BUNDLE_GENRE_ID, ((Genre) item).getId());
+                bundle.putString(Constant.BUNDLE_SEARCH_KEYWORD, ((Genre) item).getName());
+                mNavigator.startActivity(SearchResultActivity.class, bundle);
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/screen/movie_detail/MovieDetailViewModel.java]: u can put these line (117-121) to one method "redirectSearchResultScreen" with a param ``(Object item)``
It clearly and avoid duplicate code in this file

File: app/src/main/res/drawable/flat_button_light.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:enterFadeDuration="400"
+    android:exitFadeDuration="400">
+
+    <item android:drawable="@color/white_alpha" android:state_pressed="true"/>
+    <item android:drawable="@android:color/transparent"/>
+
+</selector>
```
Comment (daolq3012) [inline app/src/main/res/drawable/flat_button_light.xml]: enter when end of file

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 7]
```diff
@@ -12,84 +12,197 @@
...
-            android:layout_alignParentLeft="true"
-            android:layout_alignParentStart="true"
-            android:layout_below="@+id/image_backdrop"
-            android:gravity="center_horizontal"
-            android:text="@{viewModel.tittle}"
-            android:textColor="@color/color_tuatara"
-            android:textSize="@dimen/sp_16"
-            />
+            <TextView
+                android:id="@+id/tittle"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: pls check Framgia Convention declare name of id for view

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 8]
```diff
@@ -12,84 +12,197 @@
...
+                app:tagPaddingBottom="@dimen/dp_8"
+                app:tagPaddingLeft="@dimen/dp_6"
+                app:tagPaddingRight="@dimen/dp_6"
+                app:tagPaddingTop="@dimen/dp_8"
+                app:tagTextColor="@color/color_gray90"
+                app:tagsClickListener="@{viewModel.directorClickListener}"
+                />
+
+            <TextView
+                android:id="@+id/label_actor"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 9]
```diff
@@ -12,84 +12,197 @@
...
+                android:layout_height="wrap_content"
+                android:layout_alignParentLeft="true"
+                android:layout_alignParentStart="true"
+                android:layout_below="@+id/director"
+                android:text="@string/label_actor"
+                android:textColor="@color/color_tuatara"
+                />
+
+            <com.greenfrvr.hashtagview.HashtagView
+                android:id="@+id/actors"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 10]
```diff
@@ -12,84 +12,197 @@
...
+                android:layout_height="wrap_content"
+                android:layout_alignParentLeft="true"
+                android:layout_alignParentStart="true"
+                android:layout_below="@+id/genres"
+                android:text="@string/label_director"
+                android:textColor="@color/color_tuatara"
+                />
+
+            <com.greenfrvr.hashtagview.HashtagView
+                android:id="@+id/director"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 11]
```diff
@@ -12,84 +12,197 @@
...
+                app:tagPaddingBottom="@dimen/dp_8"
+                app:tagPaddingLeft="@dimen/dp_6"
+                app:tagPaddingRight="@dimen/dp_6"
+                app:tagPaddingTop="@dimen/dp_8"
+                app:tagTextColor="@color/color_gray90"
+                app:tagsClickListener="@{viewModel.genreClickListener}"
+                />
+
+            <TextView
+                android:id="@+id/label_director"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 12]
```diff
@@ -12,84 +12,197 @@
...
+                android:layout_height="wrap_content"
+                android:layout_alignParentLeft="true"
+                android:layout_alignParentStart="true"
+                android:layout_below="@+id/overview"
+                android:text="@string/label_genres"
+                android:textColor="@color/color_tuatara"
+                />
+
+            <com.greenfrvr.hashtagview.HashtagView
+                android:id="@+id/genres"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 13]
```diff
@@ -12,84 +12,197 @@
...
+                android:layout_height="wrap_content"
+                android:layout_alignParentLeft="true"
+                android:layout_alignParentStart="true"
+                android:layout_below="@+id/production"
+                android:text="@{viewModel.overview}"
+                android:textColor="@color/color_tuatara"
+                />
+
+            <TextView
+                android:id="@+id/label_genres"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 14]
```diff
@@ -12,84 +12,197 @@
...
+                android:layout_height="wrap_content"
+                android:layout_alignParentLeft="true"
+                android:layout_alignParentStart="true"
+                android:layout_below="@+id/vote_average"
+                android:text="@{viewModel.production}"
+                android:textColor="@color/color_tuatara"
+                />
+
+            <TextView
+                android:id="@+id/overview"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 15]
```diff
@@ -12,84 +12,197 @@
...
+                android:layout_height="wrap_content"
+                android:layout_alignParentLeft="true"
+                android:layout_alignParentStart="true"
+                android:layout_below="@+id/vote_count"
+                android:text="@{viewModel.voteAverage}"
+                android:textColor="@color/color_tuatara"
+                />
+
+            <TextView
+                android:id="@+id/production"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 16]
```diff
@@ -12,84 +12,197 @@
...
-            android:id="@+id/text_overview"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_alignParentLeft="true"
-            android:layout_alignParentStart="true"
-            android:layout_below="@+id/text_production"
-            android:text="@{viewModel.overview}"
-            />
+            <TextView
+                android:id="@+id/vote_average"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 17]
```diff
@@ -12,84 +12,197 @@
...
-            android:id="@+id/text_production"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_alignParentLeft="true"
-            android:layout_alignParentStart="true"
-            android:layout_below="@+id/text_release_date"
-            android:text="@{viewModel.production}"
-            />
+            <TextView
+                android:id="@+id/vote_count"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 18]
```diff
@@ -12,84 +12,197 @@
...
-            android:id="@+id/text_release_date"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_alignParentLeft="true"
-            android:layout_alignParentStart="true"
-            android:layout_below="@+id/text_genres"
-            android:text="@{viewModel.releaseDate}"
-            />
+            <TextView
+                android:id="@+id/release_date"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 19]
```diff
@@ -12,84 +12,197 @@
...
-            android:id="@+id/text_genres"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_alignParentLeft="true"
-            android:layout_alignParentStart="true"
-            android:layout_below="@+id/text_tittle"
-            android:text="@{viewModel.genres}"
-            />
+            <ImageView
+                android:id="@+id/image_favorite"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_movie_detail.xml]: .

File: app/src/main/java/com/example/tung/moviedb_07/screen/movie_detail/MovieDetailActivity.java (hunk 3 above)
Comment (tungnt0315) [inline app/src/main/java/com/example/tung/moviedb_07/screen/movie_detail/MovieDetailActivity.java]: em truyền fragmentManager qua để set youtubeFragment vào frameLayout trong movieDetailActivity.

File: app/src/main/java/com/example/tung/moviedb_07/utils/navigator/Navigator.java [hunk 20]
```diff
@@ -79,4 +63,8 @@ public void startActivityForResult(@NonNull Class<? extends Activity> clazz, Bun
     public void showMessage(String message) {
         Toast.makeText(mActivity, message, Toast.LENGTH_SHORT).show();
     }
+
+    public FragmentManager getFragmentManager() {
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/utils/navigator/Navigator.java]: pls remove this method from Navigator

File: app/src/main/java/com/example/tung/moviedb_07/utils/binding/BindingUtils.java [hunk 21]
```diff
@@ -40,4 +54,81 @@ public static void setImageSrc(final ImageView imageView, String imagePath) {
...
+                        }
+                    }
+
+                    @Override
+                    public void onInitializationFailure(YouTubePlayer.Provider provider,
+                            YouTubeInitializationResult youTubeInitializationResult) {
+                    }
+                });
+        FragmentManager fragmentManager = navigator.getFragmentManager();
+        fragmentManager.beginTransaction().add(frameLayout.getId(), youTubePlayerFragment).commit();
```
Comment (daolq3012) [inline app/src/main/java/com/example/tung/moviedb_07/utils/binding/BindingUtils.java]: pls write navigation in Navigator and use in here
``navigator.replaceYoutubeFragment(frameLayout.getId())``

File: app/src/main/res/drawable/flat_button_light.xml [hunk 22]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:enterFadeDuration="400"
```
Comment (daolq3012) [inline app/src/main/res/drawable/flat_button_light.xml]: pls define this value ``400`` in ``intergers.xml`` and use it in here

File: app/src/main/res/drawable/flat_button_light.xml [hunk 23]
```diff
@@ -0,0 +1,10 @@
...
+    android:exitFadeDuration="400">
```
Comment (daolq3012) [inline app/src/main/res/drawable/flat_button_light.xml]: like above

File: app/src/main/res/drawable/item_bg_1.xml [hunk 24]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:state_selected="false">
+        <shape android:shape="rectangle">
+
+            <solid android:color="#607D8B"/>
```
Comment (daolq3012) [inline app/src/main/res/drawable/item_bg_1.xml]: define in ``colors.xml`` and use it in here

File: app/src/main/res/drawable/item_bg_1.xml [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+            <corners android:radius="4dp"/>
+
+        </shape>
+    </item>
+
+    <item android:state_selected="true">
+        <shape android:shape="rectangle">
+
+            <solid android:color="#968800"/>
```
Comment (daolq3012) [inline app/src/main/res/drawable/item_bg_1.xml]: .

File: app/src/main/res/drawable/item_bg_1.xml [hunk 26]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:state_selected="false">
+        <shape android:shape="rectangle">
+
+            <solid android:color="#607D8B"/>
+            <corners android:radius="4dp"/>
```
Comment (daolq3012) [inline app/src/main/res/drawable/item_bg_1.xml]: define in ``dimens.xml``

**Pull request merged**